packages/components/src/components/FilterMenu.vue

```diff
@@ -0,0 +1,455 @@
+                  @click.stop="removeHiddenName(index)"
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import VFiltersForm from './FiltersForm.vue';
+import CheckIcon from '@/assets/check.svg';
+import CloseThinIcon from '@/assets/close-thin.svg';
+import FilterIcon from '@/assets/filter.svg';
+import ResetIcon from '@/assets/reset.svg';
+import {
+  SET_DECLUTTER_ON,
+  SET_ELAPSED_TIME,
+  RESET_FILTERS,
+  ADD_ROOT_OBJECT,
+  REMOVE_ROOT_OBJECT,
+  ADD_HIDDEN_NAME,
+  REMOVE_HIDDEN_NAME,
+} from '@/store/vsCode';
+
+export default {
+  name: 'v-filter-menu',
+
+  components: {
+    VFiltersForm,
+    CheckIcon,
+    CloseThinIcon,
+    FilterIcon,
+    ResetIcon,
+  },
+
+  props: {
+    filteredAppMap: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  computed: {
+    filters() {
+      return this.$store.state.filters;
+    },
+
+    limitRootEvents: {
+      get() {
+        return this.$store.state.filters.declutter.limitRootEvents.on;
+      },
+      set(value) {
+        this.$store.commit(SET_DECLUTTER_ON, { declutterProperty: 'limitRootEvents', value });
+      },
+    },
+
+    hideMediaRequests: {
+      get() {
+        return this.$store.state.filters.declutter.hideMediaRequests.on;
+      },
+      set(value) {
+        this.$store.commit(SET_DECLUTTER_ON, { declutterProperty: 'hideMediaRequests', value });
+      },
+    },
+
+    hideUnlabeled: {
+      get() {
+        return this.$store.state.filters.declutter.hideUnlabeled.on;
+      },
+      set(value) {
+        this.$store.commit(SET_DECLUTTER_ON, { declutterProperty: 'hideUnlabeled', value });
+      },
+    },
+
+    hideElapsedTimeUnder: {
+      get() {
+        return this.$store.state.filters.declutter.hideElapsedTimeUnder.on;
+      },
+      set(value) {
+        this.$store.commit(SET_DECLUTTER_ON, { declutterProperty: 'hideElapsedTimeUnder', value });
+      },
+    },
+
+    hideElapsedTime: {
+      get() {
+        return this.$store.state.filters.declutter.hideElapsedTimeUnder.time;
+      },
+      set(value) {
+        this.$store.commit(SET_ELAPSED_TIME, value);
+      },
+    },
+
+    hideName: {
+      get() {
+        return this.$store.state.filters.declutter.hideName.on;
+      },
+      set(value) {
+        this.$store.commit(SET_DECLUTTER_ON, { declutterProperty: 'hideName', value });
+      },
+    },
+
+    hiddenNames() {
+      return this.$store.state.filters.declutter.hideName.names;
+    },
+
+    rootObjectsSuggestions() {
+      return this.$store.state.appMap.classMap.codeObjects
+        .map((co) => co.fqid)
+        .filter((fqid) => !this.filters.rootObjects.includes(fqid));
+    },
+
+    hideNamesSuggestions() {
+      return this.filteredAppMap.classMap.codeObjects
+        .map((co) => co.fqid)
+        .filter((fqid) => !this.filters.declutter.hideName.names.includes(fqid));
```

Comment:
We could consider dropping `filteredAppMap` in favor of a list of code object fqids. It's a little better for componentization because it limits the amount of state required to render the component.